src/web/UserPresence.ts

```diff
@@ -18,21 +18,24 @@ if (typeof(document) !== 'undefined') {
 
 export class UserPresence extends RX.UserPresence {
     private _isPresent: boolean;
+    private _isAppFocused: boolean;
 
     constructor() {
         super();
         // Handle test environment where document is not defined.
         if (typeof(document) !== 'undefined') {
-            this._isPresent = ifvisible.now();
+            this._isPresent = this._isAppFocused = ifvisible.now();
 
             ifvisible.on('wakeup', this._handleWakeup.bind(this));
             ifvisible.on('idle', this._handleIdle.bind(this));
             ifvisible.on('focus', this._handleFocus.bind(this));
             ifvisible.on('blur', this._handleBlur.bind(this));
 
             window.addEventListener('blur', this._handleWindowBlur.bind(this));
+            window.addEventListener('focus', this._handleFocus.bind(this));
```

Comment:
nit: I think we can use the following syntax instead of explicitly use `.bind` in each callback

```typescript
window.addEventListener('focus', this._handleFocus);
```

```typescript
private _handleFocus = (): void => {
  ...
}
```

Reply:
In this case, we can't use this approach because the code is within the object's constructor. The instance variables (e.g. "_handleFocus") may not be initialized at the time the constructor runs.

Reply:
I think this comment is not valid anymore. I have removed using the same callback for two events..